testsuite/integration/src/tests/slurmctld.py

```diff
@@ -0,0 +1,109 @@
+#
+# Copyright 2022 Hewlett Packard Enterprise Development LP
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import docker
+import os
+import warnings
+from tenacity import *
+
+# Submitting jobs can fail, occasionally, when the DWS webhook rejects the
+# mutating webhook connection. This has only been observed using the
+# directive rules included with  the dws-test-driver in a local kind
+# environment. 
+class JobSubmissionError(Exception):
+    pass
+
+class Slurmctld:
+    def __init__(self):
+        self.slurmctld = docker.from_env().containers.get("slurmctld")
+
+    def exec_run(self, cmd):
+        print("Slurmctld exec_run: " + cmd)
+        exec_cmd = cmd.split()
+        rc,out = self.slurmctld.exec_run(
+            exec_cmd, 
+            user="slurm",
+            workdir="/jobs"
+        )
+        return rc,str(out, 'utf-8')
+    
+    @retry(
+        wait=wait_fixed(6),
+        stop=stop_after_attempt(10),
+        reraise=True
+    )
+    def submit_job(self,scriptPath):
+        print("submit job")
+        # The --wait option could be used here. However, other tests need to
+        # asynchronously track the job status
+        cmd = "sbatch --output=%(s)s.out --error=%(s)s.error.out %(s)s " % {"s": scriptPath}
```

Comment:
This is old-style string formatting.  We should be using the `.format()` method or f-strings.